core/host.go

```diff
@@ -7,90 +7,62 @@ import (
 	"strings"
 
 	"github.com/dagger/dagger/core/pipeline"
-	bkclient "github.com/moby/buildkit/client"
+	"github.com/dagger/dagger/core/socket"
+	"github.com/dagger/dagger/engine/buildkit"
 	"github.com/moby/buildkit/client/llb"
 	bkgw "github.com/moby/buildkit/frontend/gateway/client"
 	specs "github.com/opencontainers/image-spec/specs-go/v1"
 	"github.com/vito/progrock"
 )
 
 type Host struct {
-	Workdir   string
-	DisableRW bool
 }
 
-func NewHost(workdir string, disableRW bool) *Host {
-	return &Host{
-		Workdir:   workdir,
-		DisableRW: disableRW,
-	}
+func NewHost() *Host {
+	return &Host{}
 }
 
 type CopyFilter struct {
 	Exclude []string
 	Include []string
 }
 
-func (host *Host) Directory(ctx context.Context, gw bkgw.Client, dirPath string, p pipeline.Path, pipelineNamePrefix string, platform specs.Platform, filter CopyFilter) (*Directory, error) {
-	if host.DisableRW {
-		return nil, ErrHostRWDisabled
-	}
-
-	var absPath string
-	var err error
-	if filepath.IsAbs(dirPath) {
-		absPath = dirPath
-	} else {
-		absPath = filepath.Join(host.Workdir, dirPath)
-
-		if !strings.HasPrefix(absPath, host.Workdir) {
-			return nil, fmt.Errorf("path %q escapes workdir; use an absolute path instead", dirPath)
-		}
-	}
-
-	absPath, err = filepath.EvalSymlinks(absPath)
-	if err != nil {
-		return nil, fmt.Errorf("eval symlinks: %w", err)
-	}
+func (host *Host) Directory(
+	ctx context.Context,
+	bk *buildkit.Client,
+	dirPath string,
+	p pipeline.Path,
+	pipelineNamePrefix string,
+	platform specs.Platform,
+	filter CopyFilter,
+) (*Directory, error) {
+	// TODO: enforcement that requester session is granted access to source session at this path
```

Comment:
Is this a blocker, or OK for now assuming the status quo usage pattern hasn't changed (everyone is trusted, you can already run insecure payloads, etc etc)?

I have similar concerns with c2h networking right around the corner which allows arbitrary access to the host network.

Reply:
Yeah I'm not considering any of the enforcement stuff a blocker right now since it's only necessary in order to give Zenith the security properties we want, which is obviously important for when we release Zenith, but not for this PR. I went with the same idea for secrets, sockets and registry auth, I'd probably put c2h networking in the same bucket (just secure it before official release of zenith)